lib/colorado_lottery.rb

```diff
@@ -0,0 +1,58 @@
+class ColoradoLottery
+  attr_reader :registered_contestants, :winners,
+              :current_contestants
+
+  def initialize
+    @registered_contestants = {}
+    @winners = []
+    @current_contestants = {}
+  end
+
+  def interested_and_18?(contestant, game)
+    if contestant.age < 18
+      false
+    elsif !contestant.game_interests.include?(game.name)
+      false
+    else
+      true
+    end
+  end
+
+  def can_register?(contestant, game)
+    if !interested_and_18?(contestant, game)
+      false
+    elsif contestant.out_of_state? && !game.national_drawing?
+      false
+    else
+      true
+    end
+  end
+
+  def register_contestant(contestant, game)
+    if can_register?(contestant, game) &&
+       @registered_contestants[game.name].nil?
+      @registered_contestants[game.name] = [contestant]
+    elsif can_register?(contestant, game)
+      @registered_contestants[game.name] << contestant
+    end
```

Comment:
this is good-- maybe a slight cleanup here to avoid the elseif could be:

```
def register_contestant(contestant, game)
    if registered_contestants[game.name].nil?
      registered_contestants[game.name] = []
    end
    registered_contestants[game.name] << contestant
  end
```